docs/src/managing-packages.md

```diff
@@ -133,6 +133,38 @@ Updating `~/.julia/environments/v1/Manifest.toml`
   [92a963f6] + ImportMacros v1.0.0 `git@github.com:fredrikekre/ImportMacros.jl.git#master`
 ```
 
+### Adding a subdir package
+
+If the package you want to add is not in the root of the repository, then you need to manually pass the `subdir` keyword to `Pkg.add`
+or `PackageSpec`. For instance, to add the `SnoopCompileCore` package in the [SnoopCompile](https://github.com/timholy/SnoopCompile.jl)
+repository:
+
+```julia-repl
+julia> Pkg.add(url="https://github.com/timholy/SnoopCompile.jl.git", subdir="SnoopCompileCore")
+     Cloning git-repo `https://github.com/timholy/SnoopCompile.jl.git`
+    Updating git-repo `https://github.com/timholy/SnoopCompile.jl.git`
+   Resolving package versions...
+  Installing known registries into `~/.julia`
+       Added registry `General` to `~/.julia/registries/General`
+    Updating `~/.julia/environments/v1.6/Project.toml`
+  [e2b509da] + SnoopCompileCore v2.7.0 `https://github.com/timholy/SnoopCompile.jl.git:SnoopCompileCore#master`
+    Updating `~/.julia/environments/v1.6/Manifest.toml`
+  [e2b509da] + SnoopCompileCore v2.7.0 `https://github.com/timholy/SnoopCompile.jl.git:SnoopCompileCore#master`
+  [9e88b42a] + Serialization
+Precompiling project...
+  1 dependency successfully precompiled in 2 seconds
+```
+
+Another way is to use the Pkg REPL with `<repo_url>:<subdir>` format:
+
+```julia-repl
+(@v1.6) pkg> add https://github.com/timholy/SnoopCompile.jl.git:SnoopCompileCore
+...
+```
+
+!!! compat "Julia 1.6"
+    The Pkg REPL for subdir packages requires at least Julia 1.6.
```

Comment:
I'm fairly certain that this was in place for Julia 1.5.

Reply:
Hmmm...

```julia
julia> versioninfo()
Julia Version 1.5.4
Commit 69fcb5745b (2021-03-11 19:13 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin18.7.0)
  CPU: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)

(@v1.5) pkg> add https://github.com/timholy/SnoopCompile.jl.git:SnoopCompileCore
   Updating git-repo `https://github.com/timholy/SnoopCompile.jl.git`
ERROR: failed to fetch from https://github.com/timholy/SnoopCompile.jl.git, error: GitError(Code:EINVALIDSPEC, Class:Net, invalid url, missing path)
```

Reply:
Looks like #2054?

Reply:
Seems not. I'll keep this untouched until someone knows what happens here.

Reply:
```
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.5.0 (2020-08-01)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

(@v1.5) pkg> activate --temp
 Activating new environment at `/tmp/jl_qq7ekh/Project.toml`

(jl_qq7ekh) pkg> add https://github.com/timholy/SnoopCompile.jl.git:SnoopCompileCore
    Cloning git-repo `https://github.com/timholy/SnoopCompile.jl.git`
   Updating git-repo `https://github.com/timholy/SnoopCompile.jl.git`
   Updating registry at `~/.julia/registries/General`
   Updating git-repo `https://github.com/JuliaRegistries/General.git`
  Resolving package versions...
Updating `/tmp/jl_qq7ekh/Project.toml`
  [e2b509da] + SnoopCompileCore v2.7.0 `https://github.com/timholy/SnoopCompile.jl.git:SnoopCompileCore#master`
Updating `/tmp/jl_qq7ekh/Manifest.toml`
  [e2b509da] + SnoopCompileCore v2.7.0 `https://github.com/timholy/SnoopCompile.jl.git:SnoopCompileCore#master`
  [9e88b42a] + Serialization
```

Reply:
There seems to be a bug on macOS Julia 1.5, I've tried it again on the Julia docker container(`docker run --rm -it julia:1.5.4`) and it's working well.

Since it's working well for macOS Julia 1.6, I'm not going to file an bug report for this (guess there won't be any new Julia 1.5.x releases)